# Project 1 — Student Grade Analyzer

## Problem Statement
Analyzed student marks data to assign grades, determine 
pass/fail status, and summarize overall class performance.

## Dataset
- Total Students: 10
- Subjects: Math, Science, English
- Marks Range: 0 to 100

## Goals
- Assign grades to each student (A, B, C, D, F)
- Identify pass/fail status
- Calculate class average, highest and lowest marks
- Count total pass and fail students

In [1]:
students = {
    "Ali":     {"marks": 92, "subject": "Math"},
    "Sara":    {"marks": 76, "subject": "Science"},
    "Hamza":   {"marks": 55, "subject": "Math"},
    "Zara":    {"marks": 88, "subject": "English"},
    "Bilal":   {"marks": 63, "subject": "Science"},
    "Hina":    {"marks": 45, "subject": "English"},
    "Usman":   {"marks": 71, "subject": "Math"},
    "Fatima":  {"marks": 95, "subject": "Science"},
    "Kamran":  {"marks": 38, "subject": "English"},
    "Nida":    {"marks": 80, "subject": "Math"},
}

In [2]:
for name in students:

    marks = students[name]["marks"]

    if marks >= 90:
        print(name, 'A grade')

    elif 80 <= marks < 90:
        print(name, 'B grade')

    elif 70 <= marks < 80:
        print(name, 'C grade')

    elif 60 <= marks < 70:
        print(name, 'D grade')

    else:
        print(name, 'Fail')
        

Ali A grade
Sara C grade
Hamza Fail
Zara B grade
Bilal D grade
Hina Fail
Usman C grade
Fatima A grade
Kamran Fail
Nida B grade


In [3]:
marks_list = []

for name in students:
    marks_list.append(students[name]["marks"])

print("Average =", sum(marks_list) / len(marks_list))
print("Highest =", max(marks_list))
print("Lowest =", min(marks_list))

Average = 70.3
Highest = 95
Lowest = 38


In [4]:
pass_count = 0
fail_count = 0
for name in students:
    marks = students[name]["marks"]
    if marks >= 60:
         pass_count = pass_count+1
         
        
        
    else:
        fail_count = fail_count+1
print('pass',pass_count)
print('fail',fail_count)
        

pass 7
fail 3


In [5]:
students = {
    "Ali":     {"marks": 92, "subject": "Math"},
    "Sara":    {"marks": 76, "subject": "Science"},
    "Hamza":   {"marks": 55, "subject": "Math"},
    "Zara":    {"marks": 88, "subject": "English"},
    "Bilal":   {"marks": 63, "subject": "Science"},
    "Hina":    {"marks": 45, "subject": "English"},
    "Usman":   {"marks": 71, "subject": "Math"},
    "Fatima":  {"marks": 95, "subject": "Science"},
    "Kamran":  {"marks": 38, "subject": "English"},
    "Nida":    {"marks": 80, "subject": "Math"},
}

pass_count = 0
fail_count = 0
marks_list = []

print("===== GRADE REPORT =====")

for name in students:

    marks = students[name]["marks"]
    marks_list.append(marks)

    if marks >= 90:
        grade = "A"
    elif marks >= 80:
        grade = "B"
    elif marks >= 70:
        grade = "C"
    elif marks >= 60:
        grade = "D"
    else:
        grade = "F"

    if marks >= 60:
        status = "Pass"
        pass_count += 1
    else:
        status = "Fail"
        fail_count += 1

    print(f"{name:<10} | {marks} marks | {grade} grade | {status}")

average_marks = sum(marks_list) / len(marks_list)
highest_marks = max(marks_list)
lowest_marks = min(marks_list)

print("=" * 24)
print("Total Students :", len(students))
print("Pass           :", pass_count)
print("Fail           :", fail_count)
print("Class Average  :", round(average_marks, 1))
print("Highest Marks  :", highest_marks)
print("Lowest Marks   :", lowest_marks)

===== GRADE REPORT =====
Ali        | 92 marks | A grade | Pass
Sara       | 76 marks | C grade | Pass
Hamza      | 55 marks | F grade | Fail
Zara       | 88 marks | B grade | Pass
Bilal      | 63 marks | D grade | Pass
Hina       | 45 marks | F grade | Fail
Usman      | 71 marks | C grade | Pass
Fatima     | 95 marks | A grade | Pass
Kamran     | 38 marks | F grade | Fail
Nida       | 80 marks | B grade | Pass
Total Students : 10
Pass           : 7
Fail           : 3
Class Average  : 70.3
Highest Marks  : 95
Lowest Marks   : 38


## Conclusion
- 7 out of 10 students passed (70% pass rate)
- Class average was 70.3 out of 100
- Fatima scored highest (95) — Grade A
- Kamran scored lowest (38) — Grade F
- Only 20% students achieved Grade A, 
  suggesting room for improvement